podcasts/New Search/SearchResultsViewController.swift

```diff
@@ -1,12 +1,44 @@
 import Foundation
 import SwiftUI
 
-class SearchResultsViewController: OnboardingHostingViewController<AnyView> {
+protocol SearchResultsDelegate {
+    func clearSearch()
+    func performLocalSearch(searchTerm: String)
+    func performRemoteSearch(searchTerm: String, completion: @escaping (() -> Void))
+    func performSearch(searchTerm: String, triggeredByTimer: Bool, completion: @escaping (() -> Void))
```

Comment:
> I'm curious if we need all of these since it seems like there's some overlap between them, like performRemoteSearch vs performSearch?

The plan moving forward is to have one function to clear results and one to perform a search — independently from where it comes.

I just don't want to get ahead of myself and improve the current search without having the new one in place (given the current will be ditched anyway).

Plus, who should control remote results, or not should be the search entity, not the search bar as it is now.

Are you ok with changing that in the future?